### PR TITLE
YD-109 HTML email rendering by template

### PR DESCRIPTION
**Pull request by 2gmpoyhonen**

### CODE REVIEW

File: core/src/main/java/nu/yona/server/email/EmailService.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+				MimeMessageHelper message = new MimeMessageHelper(mimeMessage);
+				message.setFrom(new InternetAddress(senderAddress, senderName));
+				message.setTo(receiverAddress);
+				message.setSubject(subjectText);
+				message.setText(bodyText, true);
+			}
+		};
+
+		// TODO queue and continue because it may take time to send
```
Comment (Bert-R) [inline core/src/main/java/nu/yona/server/email/EmailService.java]: Please create a ticket for this and mention that here.

File: core/src/main/java/nu/yona/server/email/EmailService.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	@Autowired
+	private VelocityEngine velocityEngine;
+	@Value("${yona.email.sender.address}")
+	private String senderAddress;
+
+	public void sendEmail(String senderName, InternetAddress receiverAddress, String subjectTemplateName, String bodyTemplateName,
+			Map<String, Object> templateParameters)
+	{
+		MimeMessagePreparator preparator = new MimeMessagePreparator() {
+			public void prepare(MimeMessage mimeMessage) throws Exception
```
Comment (Bert-R) [inline core/src/main/java/nu/yona/server/email/EmailService.java]: Remove the ``throws`` here and put the try/catch of ``BuddyService`` inside.

File: core/src/main/java/nu/yona/server/email/EmailService.java (hunk 1 above)
Comment (2gmpoyhonen) [inline core/src/main/java/nu/yona/server/email/EmailService.java]: I have created task http://jira.yona.nu/browse/YD-114 for this. I also mention that in the comment.

Comment (aristotelos) [inline core/src/main/java/nu/yona/server/email/EmailService.java]: I added this TODO, but on further consideration I think it is not necessary. It is OK if sending the email takes a little longer, and if it fails it is better to have that info in the app directly as Philip pointed out.

Comment (Bert-R) [inline core/src/main/java/nu/yona/server/email/EmailService.java]: Agreed. I'll remove it, including the ticket.

**Comment by Bert-R** [comment]
As I'm preparing the demo scenario for Tuesday today, I'll merge the property changes into this, address the review comments and pull the change set.

**Pull request merged**